src/app-gocardless/services/gocardless-service.js

```diff
@@ -35,24 +35,26 @@ const getGocardlessClient = () => {
   return clients.get(hash);
 };
 
-export const handleGoCardlessError = (response) => {
-  switch (response.status_code) {
+export const handleGoCardlessError = (error) => {
+  const status = error?.response?.status;
+
+  switch (status) {
     case 400:
-      throw new InvalidInputDataError(response);
+      throw new InvalidInputDataError(error);
     case 401:
-      throw new InvalidGoCardlessTokenError(response);
+      throw new InvalidGoCardlessTokenError(error);
     case 403:
-      throw new AccessDeniedError(response);
+      throw new AccessDeniedError(error);
     case 404:
-      throw new NotFoundError(response);
+      throw new NotFoundError(error);
     case 409:
-      throw new ResourceSuspended(response);
+      throw new ResourceSuspended(error);
     case 429:
-      throw new RateLimitError(response);
+      throw new RateLimitError(error);
     case 500:
-      throw new UnknownError(response);
+      throw new UnknownError(error);
     case 503:
-      throw new ServiceError(response);
+      throw new ServiceError(error);
     default:
       return;
```

Comment:
Now that we explicitly pass in `error`, the `default` case should throw a generic error, not silently return